src/eterna/constraints/Constraint.ts

```diff
@@ -15,6 +15,15 @@ export interface HighlightInfo {
     stateIndex?: number;
 }
 
+export interface ConstraintContext {
+    undoBlocks: UndoBlock[];
+    targetConditions?: any[];
+    puzzle?: Puzzle;
+    targetPairs?: number[][];
+    currentTargetIndex?: number;
+    sequence?: number[];
```

Comment:
One last thing to comment on here. targetPairs and sequence are already on undoBlock, so you don't need to pass them separately. Also, currentTargetIndex should not have an impact on how the constraint is evaluated - if a constraint doesn't have a parameter specifying the state, that means currently we only support it in state 1 (ie, index 0) so you can hardcode that in for the boost constraint.

Reply:
OK will do. Note that Nova pulls both `targetPairs` and `sequence` from PoseEdit state, from what you're saying, I assume `undoBlock[0]` is pointing to the same thing.

Reply:
Done